azure/job.py

```diff
@@ -0,0 +1,96 @@
+import datetime
+import sys
+import os
+import uuid
+
+from azure.identity import DefaultAzureCredential
+from msrest.authentication import BasicTokenAuthentication
+from azure.storage.blob import BlobServiceClient
+from azure.batch import BatchServiceClient
+import azure.batch.models as batchmodels
+
+blob_account = os.environ["BLOB_ACCOUNT"]
+blob_url = f"https://{blob_account}.blob.core.windows.net"
+batch_account = os.environ["BATCH_ACCOUNT"]
+batch_url = f"https://{batch_account}.eastus.batch.azure.com"
+config_container = sys.argv[1]
+pool_id = sys.argv[2]
+# Re-use Azure Pool name unless otherwise specified
+job_id = sys.argv[3] if len(sys.argv) > 3 else pool_id
+
+if __name__ == "__main__":
+    # Authenticate with workaround because Batch is the one remaining
+    # service that doesn't yet support Azure auth flow v2 :) :)
+    # https://github.com/Azure/azure-sdk-for-python/issues/30468
+    credential_v2 = DefaultAzureCredential()
+    token = {"access_token": credential_v2.get_token("https://batch.core.windows.net/.default").token}
+    credential_v1 = BasicTokenAuthentication(token)
+
+    batch_client = BatchServiceClient(
+        credentials=credential_v1,
+        batch_url=batch_url
+     )
+
+    #############
+    # Set up job
+    job = batchmodels.JobAddParameter(
+        id=job_id,
+        pool_info=batchmodels.PoolInformation(pool_id=pool_id)
+    )
+
+    try:
+        batch_client.job.add(job)
+    except batchmodels.BatchErrorException as err:
+        if err.error.code != "JobExists":
+            raise
+        else:
+            print("Job already exists. Using job object")
+
+    ##########
+    # Get tasks
+    blob_service_client = BlobServiceClient(blob_url, credential_v2)
+    container_client = blob_service_client.get_container_client(container=config_container)
+    two_mins_ago = datetime.datetime.now(datetime.UTC) - datetime.timedelta(minutes=2)
+    task_configs = []
+    blobs = container_client.list_blobs()
+
+    for blob in blobs:
+        if blob.creation_time > two_mins_ago:
+            task_configs.append(blob.name)
```

Comment:
```suggestion
    task_configs: list[str] = [
        b.name
        for b in container_client.list_blobs()
        if b.creation_time > two_mins_ago
    ]
```

Feel free to ignore. Just adding bc I like list comprehensions.